docs/guide/migration.md

```diff
@@ -27,27 +27,39 @@ A small fraction of users will now require using [@vitejs/plugin-legacy](https:/
   - `build.polyfillDynamicImport` (use [`@vitejs/plugin-legacy`](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy) for browsers without dynamic import support)
   - `optimizeDeps.keepNames` (switch to [`optimizeDeps.esbuildOptions.keepNames`](../config/dep-optimization-options.md#optimizedepsesbuildoptions))
 
-## Dev Server Changes
+## Achitecture changes and legacy Options
+
+This section describes the biggest architecture changes in Vite v3. To allow projects to migrate from v2 in case of a compat issue, lecagy options have been added to revert to the Vite v2 strategies.
+
+:::warning
+These options are marked as experimental and deprecated. They may be removed in a future v3 minor without respecting semver. Please pin the Vite version when using them.
+
+- `legacy.devDepsScanner`
+- `legacy.buildRollupPluginCommonjs`
+- `legacy.buildSsrCjsExternalHeuristics`
+  :::
```

Comment:
```suggestion

:::
```

It seems it need to be like this because `::` will be inside the list if it is placed in the next line. 😳